### PR TITLE
Customer Information

### PR DESCRIPTION
**Pull request by caressahh**
## Pt Link: https://www.pivotaltracker.com/story/show/150510531 ##

- Added fields for customer information

- payment button is disabled until all fields are filled out, then it is enabled

- message is visable ( requiring the cutomer to fill in information) until the fields are filled in

## things i've learned ##

-  how to work with field requirements

- how so disable abd enable a button/message with conditions

### CODE REVIEW

**Comment by tochman** [comment]
I would really suggest that you have a user story and a feature that justifies creating a customer model.

**Comment by caressahh** [comment]
Tests being created now, i was just testing if the jquery worked and shouldnt have pushed,,,

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -7,6 +7,7 @@
     <%= javascript_include_tag "https://maps.googleapis.com/maps/api/js" %>
     <%= javascript_include_tag "https://cdnjs.cloudflare.com/ajax/libs/gmaps.js/0.4.25/gmaps.min.js" %>
     <%= javascript_include_tag 'application', data:{turbolinks: {track: 'reload'}} %>
+    <%= javascript_include_tag "https://code.jquery.com/jquery-3.2.1.min.js", integrity: "sha256-hwg4gsxgFZhOsEEamdOYGBf13FyQuiTwlAQgxVSNgt4=", crossorigin: "anonymous"%>
```
Comment (tochman) [inline app/views/layouts/application.html.erb]: I don't think you need to include jQuery. You guys have the jQuery gem in Gemfile?

File: app/views/orders/show.html.erb [hunk 2]
```diff
@@ -8,17 +8,54 @@
...
+
 <div>Total price: <%= @order.total %></div>
 <%= form_tag charges_path do %>
+  <div>Name: <%= text_field_tag 'name', '', onblur: 'checkMandatoryValues()'%></div>
+  <div>Address 1: <%= text_field_tag 'address1', '', onblur: 'checkMandatoryValues()' %></div>
+  <div>Address 2: <%= text_field_tag 'address2' %></div>
+  <div>Postal Code: <%= text_field_tag 'postal_code', '', onblur: 'checkMandatoryValues()' %></div>
+  <div>City: <%= text_field_tag 'city', '', onblur: 'checkMandatoryValues()' %></div>
+  <div>E-mail: <%= text_field_tag 'email', '', onblur: 'checkMandatoryValues()' %></div>
+  <div>Phone: <%= text_field_tag 'phone_number', '', onkeyup: 'checkMandatoryValues()' %></div>
```
Comment (tochman) [inline app/views/orders/show.html.erb]: Is Materialize added? Perhaps you should use the framework to  style the form? Also, please use `label_for` for the labels

File: app/views/orders/show.html.erb [hunk 3]
```diff
@@ -8,17 +8,54 @@
...
+
   <% if flash[:error].present? %>
     <div id="error_explanation">
       <p><%= flash[:error] %></p>
     </div>
   <% end %>
 
+  <div id="mandatoryAddressMessage">Customer Information Must be Filled in</div>
```
Comment (tochman) [inline app/views/orders/show.html.erb]: `mandatoryAddressMessage` should be in other format  -> `mandatory-address-message`

File: features/missing_customer_info.feature [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  Scenario: I check out and pay with my credit card
+    When I visit "restaurant" page
+    And I add "papadom" to order
+    And I click on "Show Order"
+    Then I should be on the "order" page
+    And I should see "Address 1"
+    And the "Pay with Card" button should be disabled
+    Then I should see "Customer Information Must be Filled in"
+    And I should be on the "order" page
```
Comment (tochman) [inline features/missing_customer_info.feature]: And now add a scenario where the customer is adding the necessary info and is able to see an enabled stripe button

**Review by tochman — CHANGES_REQUESTED** [review]

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (caressahh) [reply app/views/layouts/application.html.erb]: yeah... for some reason my code didn't work until i added it there.... i will remove and try again

File: app/views/orders/show.html.erb (hunk 3 above)
Comment (caressahh) [reply app/views/orders/show.html.erb]: dangit! i'll get the hang of it eventually... fixing now

File: features/missing_customer_info.feature (hunk 4 above)
Comment (caressahh) [reply features/missing_customer_info.feature]: done!

**Review by tochman — APPROVED** [review]

**Pull request merged**